pkg/types/ccipocr3/generic_types.go

```diff
@@ -82,51 +82,85 @@ func (c ChainSelector) String() string {
-	Metadata CCIPMsgMetadata `json:"metadata"`
-}
-
-type CCIPMsgMetadata struct {
-	// Version of the message metadata. Required in order to be able to parse the metadata
-	// by the underlying implementation.
-	Version string `json:"version"`
-	// Data is the metadata payload. The underlying implementation should know how to parse this data.
-	Data []byte `json:"data"`
-}
-
-type TokenAmount struct {
-	Token  types.Account
-	Amount *big.Int
-}
-
-func (c CCIPMsg) String() string {
+// Message is the generic Any2Any message type for CCIP messages.
+// It represents, in particular, a message emitted by a CCIP onramp.
+// The message is expected to be consumed by a CCIP offramp after
+// translating it into the appropriate format for the destination chain.
+type Message struct {
+	// Header is the family-agnostic header for OnRamp and OffRamp messages.
+	// This is always set on all CCIP messages.
+	Header RampMessageHeader `json:"header"`
+
+	// EVM2AnyRampMessage is populated if the CCIP message originated from an EVM chain.
+	EVM2AnyRampMessage *EVM2AnyRampMessage `json:"evm2AnyRampMessage,omitempty"`
+}
+
+func (c Message) String() string {
 	js, _ := json.Marshal(c)
 	return string(js)
 }
 
-type CCIPMsgBaseDetails struct {
-	// ID is a unique identifier for the message, it should be unique across all chains.
+// RampMessageHeader is the family-agnostic header for OnRamp and OffRamp messages.
+// The MessageID is not expected to match MsgHash, since it may originate from a different
+// ramp family.
+type RampMessageHeader struct {
+	// MessageID is a unique identifier for the message, it should be unique across all chains.
 	// It is generated on the chain that the CCIP send is requested (i.e. the source chain of a message).
-	ID string `json:"id"`
-	// SourceChain is the chain that the message originated from.
-	SourceChain ChainSelector `json:"sourceChain,string"`
-	// SeqNum is an auto-incrementing sequence number for the message.
-	// NOTE: Sequence numbers are unique per chain. Meaning that the same sequence number can exist on multiple chains.
-	SeqNum SeqNum `json:"seqNum,string"`
+	MessageID Bytes32 `json:"messageId"`
+	// SourceChainSelector is the chain selector of the chain that the message originated from.
+	SourceChainSelector ChainSelector `json:"sourceChainSelector,string"`
+	// DestChainSelector is the chain selector of the chain that the message is destined for.
+	DestChainSelector ChainSelector `json:"destChainSelector,string"`
+	// SequenceNumber is an auto-incrementing sequence number for the message.
+	// Not unique across lanes.
+	SequenceNumber SeqNum `json:"seqNum,string"`
+	// Nonce is the nonce for this lane for this sender, not unique across senders/lanes
+	Nonce uint64 `json:"nonce"`
 
 	// MsgHash is the hash of all the message fields.
 	// NOTE: The field is expected to be empty, and will be populated by the plugin using the MsgHasher interface.
 	MsgHash Bytes32 `json:"msgHash"` // populated
```

Comment:
Is the destination's commit-root computed based on these hashes? If so, the on-chain code should recompute them and use the commit-root as a CRC on the parameters. This would ensure the message fields haven't been tampered with since the commit report was created.

```suggestion
	MsgHash Bytes32 `json:"-"` // populated
```

Reply:
> Is the destination's commit-root computed based on these hashes?

Yes, the dest commit root will be computed by first "translating" the Any2Any message into Any2DestFamily and then hashing it after encoding it appropriately in the dest family specific encoding scheme + hash function.

> If so, the on-chain code should recompute them and use the commit-root as a CRC on the parameters. This would ensure the message fields haven't been tampered with since the commit report was created.

The onchain code will receive a Any2DestFamily message, re-hash it, and use the provided proof to verify that it was part of the merkle root committed to in the commit phase. If message data was tampered with the proof would not verify onchain.

Reply:
Thanks, that was my understanding. In that case this field isn't needed onchain except for caching or potentially failing the transaction earlier than when validating against the recomputed merkle root.